app/controllers/application_controller.rb

```diff
@@ -93,12 +93,29 @@ def check_syndicated_layout
     'syndicated' if syndicated_tool_request?
   end
 
-  def category_tree
-    @category_tree ||= Core::CategoryTreeReader.new.call
+  def category_tree(categories = Core::Registry::Repository[:category].all)
+    Core::CategoryTreeReader.new.call(categories)
   end
 
-  def category_navigation
-    @category_navigation ||= CategoryNavigationDecorator.decorate_collection(category_tree.children)
+  def navigation_categories
+    Core::Registry::Repository[:category].all
+  end
+
+  def corporate_categories
+    Core::Registry::Repository[:category].find('corporate-categories')['contents']
+  end
+
+  def corporate_category?(category, corporate = corporate_categories, categories = [])
```

Comment:
This method is listed as a private method, but is also declared as a helper_method. Making it private to the class but publicly available as a view helper. Intentional?

The method is doing a number of different things (including being recursive) which I would be a bit uncomfortable changing without having tests describing the existing behaviour.